ykcompile/src/lib.rs

```diff
@@ -194,6 +212,7 @@ impl TraceCompiler {
                     );
                 }
             }
+            Some(Location::Arg(_)) => {},
```

Comment:
Can this ever happen? Should it be `unreachable!`?

Reply:
To be honest, I don't know. Since only trace inputs return `Arg` I need to check if they always outlive the trace so we won't see a `StorageDead` for them. This might be the case if the `inputs` variable is not used after tracing has finished.

Reply:
A quick test showed that even if the `inputs` variable is not used after the trace it doesn't seem to die during the trace, though I wouldn't take that as a guarantee that this can never happen. If it does happen I'm not quite sure what the right solution is.

Reply:
Shall we make it something which crashes and we can address it later if it comes up?

Reply:
Fine with me. I'll just add `unreachable` to this and the other two cases.